client/my-sites/people/followers-list/index.jsx

```diff
@@ -16,290 +16,286 @@ import PeopleListItem from 'calypso/my-sites/people/people-list-item';
+					}
 				);
-			} else {
-				followers = this.renderPlaceholders();
 			}
 
-			const downloadListLink =
-				this.props.fetchOptions.type === 'email' && !! this.props.site
-					? 'https://dashboard.wordpress.com/wp-admin/index.php?page=stats&blog=' +
-					  this.props.site.ID +
-					  '&blog_subscribers=csv&type=email'
-					: null;
+			const infiniteListConditionals = {
+				fetchingNextPage: this.props.fetching,
+				lastPage: this.isLastPage(),
+			};
 
-			return (
-				<div>
-					<PeopleListSectionHeader
-						isFollower
-						isPlaceholder={ this.props.fetching || this.props.fetchOptions.search }
-						label={ headerText }
-						site={ this.props.site }
-					>
-						{ downloadListLink && (
-							<Button href={ downloadListLink } compact>
-								{ this.props.translate( 'Download Data as CSV' ) }
-							</Button>
-						) }
-					</PeopleListSectionHeader>
-					<Card className={ listClass }>{ followers }</Card>
-					{ this.isLastPage() && <ListEnd /> }
-				</div>
+			followers = (
+				<InfiniteList
+					{ ...infiniteListConditionals }
+					key={ key }
+					items={ this.props.followers }
+					className="followers-list__infinite is-people"
+					ref={ this.infiniteList }
+					fetchNextPage={ this.fetchNextPage }
+					getItemRef={ this.getFollowerRef }
+					renderLoadingPlaceholders={ this.renderPlaceholders }
+					renderItem={ this.renderFollower }
+					guessedItemHeight={ 126 }
+				/>
 			);
+		} else {
+			followers = this.renderPlaceholders();
 		}
+
+		const downloadListLink =
+			this.props.query.type === 'email' && !! this.props.site
+				? 'https://dashboard.wordpress.com/wp-admin/index.php?page=stats&blog=' +
+				  this.props.site.ID +
+				  '&blog_subscribers=csv&type=email'
+				: null;
+
+		return (
+			<>
+				<PeopleListSectionHeader
+					isFollower
+					isPlaceholder={ this.props.fetching || this.props.query.search }
+					label={ headerText }
+					site={ this.props.site }
+				>
+					{ downloadListLink && (
+						<Button href={ downloadListLink } compact>
+							{ this.props.translate( 'Download Data as CSV' ) }
+						</Button>
+					) }
+				</PeopleListSectionHeader>
+				<Card className={ listClass }>{ followers }</Card>
+				{ this.isLastPage() && <ListEnd /> }
+			</>
+		);
 	}
-);
+}
 
-const FollowersList = ( props ) => {
-	let DataComponent;
-	const fetchOptions = {
-		max: 100,
-		page: 1,
-		search: props.search && props.search,
-		siteId: props.site.ID,
+const mapStateToProps = ( state, ownProps ) => {
+	return {
+		followers: getFollowersForQuery( state, ownProps.query ),
+		fetching: getIsFetchingFollowersForQuery( state, ownProps.query ),
+		totalFollowers: getTotalFollowersForQuery( state, ownProps.query ),
 	};
+};
 
-	if ( 'email' === props.type ) {
-		DataComponent = EmailFollowersData;
-		fetchOptions.type = 'email';
-	} else {
-		DataComponent = FollowersData;
-	}
+const ConnectedFollowers = connect( mapStateToProps, { removeFollower } )( localize( Followers ) );
```

Comment:
Should we move this component to a separate file? Or should we merge `FollowersList` and `Followers`? In general, what's the idea behind having them as 2 separate components?

Reply:
I split them up in d46e1b74a8. I think the reason why these are separate are because the `<Followers />` component renders differently depending on state.